creator/ingest_runs/genomic_data_loader.py

```diff
@@ -0,0 +1,397 @@
+            f"links in {DATASERVICE_URL}"
+        )
+        _ = self._load_entities(SEQ_EXP_GEN_FILE, df)
+
+        return se_gf_df
+
+    def _prep_harmonized_gf_df(self, df):
+        """
+        Part of Step 1
+
+        Clean up the genomic file DataFrame (output from get_s3_file_metadata)
+        and get it ready to be ingested by transforming column values and
+        standardizing the column names.
+        """
+        self.logger.info("Preparing genomic file DataFrame for ingest ...")
+
+        def file_ext(x):
+            """
+            Extract genomic file extension from filename
+            """
+            matches = [
+                file_ext
+                for file_ext in FILE_EXT_FORMAT_MAP
+                if x.endswith(file_ext)
+            ]
+            if matches:
+                file_ext = max(matches, key=len)
+            else:
+                file_ext = None
+
+            return file_ext
+
+        df['Hashes'] = df.apply(lambda row: {"ETag": row['ETag']}, axis=1)
+        df['urls'] = df.apply(lambda row: [row['Filepath']], axis=1)
+        df['file_type'] = df.apply(
+            lambda row: file_ext(row["Filename"]), axis=1,
+        )
+        col_rename = {
+            'KF Biospecimen ID': CONCEPT.BIOSPECIMEN.TARGET_SERVICE_ID,
+            'Data Type': CONCEPT.GENOMIC_FILE.DATA_TYPE,
+            'Filename': CONCEPT.GENOMIC_FILE.FILE_NAME,
+            'Hashes': CONCEPT.GENOMIC_FILE.HASH_DICT,
+            'Size': CONCEPT.GENOMIC_FILE.SIZE,
+            'urls': CONCEPT.GENOMIC_FILE.URL_LIST,
+            'file_type': CONCEPT.GENOMIC_FILE.FILE_FORMAT,
+            'Filepath': CONCEPT.GENOMIC_FILE.ID,
+            'Source Read': CONCEPT.GENOMIC_FILE.SOURCE_FILE,
+        }
+        df = df[list(col_rename.keys())]
+        df.rename(columns=col_rename, inplace=True)
+        df[CONCEPT.GENOMIC_FILE.HARMONIZED] = True
+        df[CONCEPT.GENOMIC_FILE.VISIBLE] = True
+
+        return df
+
+    def _get_genomic_file_kf_ids(self, ingest_cache, genomic_df):
+        """
+        Part of Step 1
+
+        Extract the genomic_file KF IDs from the ingest cache and add the IDs
+        as a new column to the genomic file DataFrame _genomic_df_.
+        Return _genomic_df_
+        """
+        self.logger.info(
+            "Fetching harmonized genomic file KF IDs from ingest cache ..."
+        )
+        cache_dict = {
+            ast.literal_eval(key)['external_id']: value
+            for key, value in ingest_cache["genomic_file"].items()
+        }
+        hash_df = pd.DataFrame(
+            list(cache_dict.items()),
+            columns=[
+                CONCEPT.GENOMIC_FILE.ID, CONCEPT.GENOMIC_FILE.TARGET_SERVICE_ID
+            ],
+        )
+        genomic_df = genomic_df.merge(
+            right=hash_df,
+            on=CONCEPT.GENOMIC_FILE.ID,
+            how='inner',
+        )
+        return genomic_df
+
+    def _get_seq_experiment_genomic_files(self, harmonized=False):
```

Comment:
Simplified this function (and renamed it) because the previous version was doing things with biospecimens and I don't think we need any biospecimen info for this